src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateDictionary.cs

```diff
@@ -806,26 +855,35 @@ public void Reset()
 
             public ModelStateNode GetNode(StringSegment subKey, bool createIfNotExists)
             {
-                if (subKey.Length == 0)
-                {
-                    return this;
-                }
-
-                var index = BinarySearch(subKey);
                 ModelStateNode modelStateNode = null;
-                if (index >= 0)
-                {
-                    modelStateNode = ChildNodes[index];
-                }
-                else if (createIfNotExists)
+                if (subKey.Length != 0)
```

Comment:
The inline return creates bigger asm as it sticks stack unwinding (series of pop instructions) where the returns happen and you get it for each return statement and the compiler doesn't optimise it back to this (though it should) so it would be a little slower. It is a pretty hot method - but happy to go either way?